src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -1530,6 +1537,54 @@ public void shouldReturnErrorResponseIfAllInstancesFailed()
     assertThat(instances.size(), is(0));
   }
 
+  @Test
+  public void canCreateAnInstanceWithNatureOfContent() throws Exception {
```

Comment:
What happens if an instance is created with a nature of content term that does not exist?

Reply:
Hi @marcjohnson-kint, there are no constrains for this, unfortunately.

In order to fix it we will need following:
- Create a separate table which will hold `instance-id` - `nature-of-content` columns (because natureOfContentIds in an Instance is array).
- then create FK for both fields;
- then create trigger for `instance` table which will write those values to the new table

Should we proceed with such a solution? It will increase original estimation of the task I think.

Reply:
@bohdan-suprun I think this is worth asking as a question on the issue, including folks like @skoczko @zburke @mkuklis so that we can decide on what to do here?

I don't know off the top of my head what we've done in the past, maybe you could investigate if any of the other array properties have this kind of protection?

If we haven't done it in other places maybe this is outside the scope of this story.

I think there is a simpler approach to providing this check, it does not necessarily need to be done in the database.

Reply:
@marcjohnson-kint, I've reviewed  mod-inventory-storage module and it seems to me that this is the general problem for Instance entity. 

There are no restrictions for the similar elements, which hold an external ID (e.g. contributor.nameTypeId/typeId/, identifiers.identifierTypeId - I was able to save an instance with ID which does not exist, in unit-test).

But in other modules, like `mod-users` I see, that there is additional java check, that do a query to DB and validates inputs (see `org.folio.rest.impl.UsersAPI#checkAllAddressTypesValid`).

Reply:
I think this is a requirements question back to Cate and Charlotte, to ask them if the API should check this specifically, for consistency with the UI (given that is a hot topic).

@bohdan-suprun Please ask this question on the issue.